src/Compilers/Core/Portable/CommandLine/CommonCommandLineParser.cs

```diff
@@ -485,30 +485,36 @@ internal static string RemoveTrailingSpacesAndDots(string path)
                 parsedArgs = newArgs;
                 return true;
             }
-        }
-
-        internal static bool IsClientArgsOption(string arg, string optionName, out bool hasValue, out string optionValue)
-        {
-            hasValue = false;
-            optionValue = null;
 
-            if (!arg.StartsWith(optionName, StringComparison.OrdinalIgnoreCase))
+            bool isClientArgsOption(string arg, string optionName, out bool hasValue, out string optionValue)
             {
-                return false;
-            }
+                hasValue = false;
+                optionValue = null;
 
-            if (arg.Length > optionName.Length && !(arg[optionName.Length] == ':' || arg[optionName.Length] == '='))
-            {
-                return false;
-            }
+                if (arg.Length == 0 || !(arg[0] == '/' || arg[0] == '-'))
+                {
+                    return false;
+                } 
+                
+                arg = arg.Substring(1);
+                if (!arg.StartsWith(optionName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
 
-            if (arg.Length > optionName.Length)
-            {
-                hasValue = true;
-                optionValue = arg.Substring(optionName.Length + 1).Trim('"');
-            }
+                if (arg.Length > optionName.Length && !(arg[optionName.Length] == ':' || arg[optionName.Length] == '='))
```

Comment:
>&& !(arg[optionName.Length] == ':' || arg[optionName.Length] == '=') [](start = 51, length = 68)

Consider nesting this part in another `if`, and put the following `if` as the else block so you don't have to redo the length check.